### PR TITLE
Feature/robot  bdi automation

### PR DESCRIPTION
**Pull request by skristem**
# Critical Changes

# Changes
Created a test case for bdi additional objects test case. Created page objects for Dataimport, Opportunity and Payment. Had to edit few keywords and as part of rearranging stuff for page object model, had to touch find matched gifts and payment allocations test cases
# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by Safiyab** [comment]
@skristem About the description of this PR: Anything under "Critical Changes" needs to go in the official Documentation and Release notes. Was this your expectation ? If not, you will have to move you text above the Critical Changes heading.

**Comment by skristem** [comment]
thanks, @Safiyab I just wanted to make note of everything that's been changed as there are a lot of changes. Moved them to changes section

File: robot/Cumulus/resources/OpportunityPageObject.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from cumulusci.robotframework.pageobjects import DetailPage
+from cumulusci.robotframework.pageobjects import pageobject
+from locators_46 import npsp_lex_locators
+
+@pageobject("Detail", "Opportunity")
+class OpportunityPage(DetailPage):
+    object_name = "Opportunity"
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/OpportunityPageObject.py]: oh yeah u r right didn't think about it. Thanks :)

File: robot/Cumulus/resources/DataImportPageObject.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    
+    def change_view_to(self,view_name):
+        """Changes the view on the object page to the selected view"""
+        self.npsp.select_object_dropdown()
+        locator=npsp_lex_locators['link'].format(view_name)
+        self.selenium.click_element(locator)
+        
+    def click(self,btn_name):
+        """Clicks button with specified name on Data Import Page"""
+        self.npsp.click_special_object_button(btn_name)
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/DataImportPageObject.py]: added wait as part of click special object button

**Comment by skristem** [comment]
@davidmreed and @Safiyab made changes u both suggested

**Comment by davisagli** [comment]
@skristem did you mean to restore the feature branch after merging for some reason?

**Comment by skristem** [comment]
@davisagli yes I am reusing the branch for all bdi automation work

**Pull request merged**